10up/publisher-media-kit
# Recreate blocks using only wp core - remove all css + Min WP Version Bump to 5.8

### Description of the Change

Aims of these changes:
WP patterns cannot contain CSS. We need to refactor using only wp core settings in order to prepare to add patterns to WP pattern library.

Actions from this change:

- Recreating the blocks using current WP core block options only. 
- Replacing old blocks with wp-core generated source code. 
- Removing all block css in favour of inlined styling from wp core settings.
- Replacing all images with public urls from openverse
- Also bumping the WP minimum required version from `5.7` to `5.8` because, after this PR, the block patterns will be broken in 5.7 or below.

Scope outlined by @jeffpaul 
See @jeffpaul for documentation if needed.




Closes #

### How to test the Change

Create the patterns in the plugin in wp and observe the output in the editor and FE. 

### Changelog Entry

> Added - WP Core only block code
> Changed - Existing block code and css
> Deprecated - n/a
> Removed - Existing block code and css
> Fixed - n/a
> Security - n/a


### Notes / Issues in recreating design using only wp core blocks settings
**Global content width**
No option change, so the narrow 650px content width restricts the design. For instance font sizes need to be decrease otherwise they'll look oversized against the narrow content width

**Group blocks** 
You can add borders but you can't limit the border stroke to only the bottom of the element - it's always a full element border. You can only control the border width and style. 

**Text elements** 
Margin - you can add positive margin values but you cannot reset core text element margins or apply negative margin values to compensate. 

**Tables**
Stripes - you can add stipes but you can't control the stripe background color value - its always grey 

**Tabs**
Heading - There is a heading expectation when you create a Tabs Block. But this heading does not have the usual Heading element controls like font-size, appearence etc. So it is best to create a core Heading element above the tabs block. But the Tabs block blank heading still renders and adds unwanted margin top and bottom. 

**Tabs Item** 
There are not styling controls on Tab Items - like modifying or removing the default borders or modifying the Tab heading padding to match a design 

**Columns** 
Style - no option to add border or box-shadow - so you can't replicate box Column box shadows from design

**Lists** - no option to replace standard HTML list item bullets with images/icons - so better to just use a 'Row' with an image and a paragraph block. 

**Row** - You can justify items but you can't align items - so you can't have a row with top aligned content, its always center aligned

### Credits

Props @username, @username2, ...


### Checklist:
